crates/amaru-ledger/src/rules/transaction/vkey_witness.rs

```diff
@@ -86,3 +86,280 @@ pub fn execute(
+        // The witness set is modified to include an invalid signature
+        let mut ctx: AssertValidationContext = load_validation_context_from_file("tests/data/transactions/preprod/44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca/invalid-signature/context.json").expect("failed to laod validation context");
+        let tx_bytes = include_bytes!("../../../tests/data/transactions/preprod/44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca/tx.cbor");
+        let transaction_body: KeepRaw<'_, MintedTransactionBody<'_>> =
+            cbor::decode(tx_bytes).expect("Failed to cbor decode transaction body");
+        let witness_set_bytes = include_bytes!("../../../tests/data/transactions/preprod/44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca/invalid-signature/witness.cbor");
+        let witness_set: WitnessSet =
+            cbor::decode(witness_set_bytes).expect("Failed to cbor decode witness set");
+
+        match super::execute(
+            &mut ctx,
+            transaction_body.original_hash(),
+            witness_set.vkeywitness.as_deref(),
+        ) {
+            Err(InvalidVKeyWitness::InvalidSignatures { invalid_witnesses }) => {
+                assert!(invalid_witnesses.len() == 1);
+                assert!(
+                    invalid_witnesses[0].position == 0
+                        && matches!(
+                            invalid_witnesses[0].element,
+                            InvalidEd25519Signature::InvalidSignature
+                        )
+                );
+            }
+            Ok(_) => panic!("Expected Err, got Ok"),
+            Err(e) => panic!("Unexpected error variant: {:?}", e),
+        }
+    }
+
+    #[test]
+    fn invalid_signature_length() {
+        // The following test relies on a real tranasction found on Preprod (44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca)
+        // The witness set is modified to include an invalid signature (due to length)
+        let mut ctx: AssertValidationContext = load_validation_context_from_file("tests/data/transactions/preprod/44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca/invalid-signature-length/context.json").expect("failed to laod validation context");
+        let tx_bytes = include_bytes!("../../../tests/data/transactions/preprod/44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca/tx.cbor");
+        let transaction_body: KeepRaw<'_, MintedTransactionBody<'_>> =
+            cbor::decode(tx_bytes).expect("Failed to cbor decode transaction body");
+        let witness_set_bytes = include_bytes!("../../../tests/data/transactions/preprod/44762542f8e2f66da2fa0d4fdf2eb82cc1d24ae689c1d19ffd7e57d038f50bca/invalid-signature-length/witness.cbor");
+        let witness_set: WitnessSet =
+            cbor::decode(witness_set_bytes).expect("Failed to cbor decode witness set");
```

Comment:
So, this is where a cute macro would be useful to avoid repeating the same boilerplate code over and over.